### DISCUSSION TITLE
How to get ALL available font styles using ImageSharp?

### DISCUSSION THREAD

**Discussion by SulfuricAcidH2SO4** [SixLabors/Fonts#389] [Q&A] [answered]
I would like to be able to access all the styles of a font.
Let's say for example I want to use the font "Bahnschrift" which comes preinstalled with Windows, with the specific style "SemiCondensed". Right now, after getting the family, I am not able to see all the available styles as doing :

`family.GetAvailableStyles()`

will just return a collection of enumerators with the fonts in the defined styles. As a consequence I am not able to load that specific style.

I checked [The documentation](https://docs.sixlabors.com/api/Fonts/SixLabors.Fonts.FontFamily.html), but have not been successful in finding any way to specify a custom named font style.

I am welcome to any suggestion, thanks!

**Comment by JimBobSquarePants** [comment, marked as answer]
Bahnschrift is a variable font which is currently unsupported by the library. 

See Issue https://github.com/SixLabors/Fonts/issues/5

We have a [WIP PR](https://github.com/SixLabors/Fonts/pull/342) to allow supporting CFF2 variable fonts but at this time no work has been started to allow support for TTF fonts.

**Reply by Allenyou1126** [reply]
Sorry to disturb, but i'm still curious about the progress of support for TTF fonts.

**Reply by JimBobSquarePants** [reply]
You mean variable fonts? TTF (non-variable) are fully supported.

**Reply by Allenyou1126** [reply]
Oh it's my fault, I mean the support of variable TTF fonts. Thanks for your kindly answer.
